src/RuleTypeBuilderExtensions.cs

```diff
@@ -0,0 +1,34 @@
+﻿using FluentValidation;
+using Microsoft.AspNetCore.Mvc;
+using System.Linq;
+
+namespace JSM.FluentValidation.AspNet.AsyncFilter
+{
+    /// <summary>
+    /// Extension to override error code with prefix and type.
+    /// </summary>
+    public static class RuleTypeBuilderExtensions
+    {
+        /// <summary>
+        /// Overrides the error code associated with this rule with prefix and type
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <typeparam name="TProperty"></typeparam>
+        /// <param name="rule">The current rule</param>
+        /// <param name="type">The type used to override error code</param>
+        /// <returns></returns>
+        public static IRuleBuilderOptions<T, TProperty> WithRuleType<T, TProperty>(this IRuleBuilderOptions<T, TProperty> rule, string type)
+        {
+            string propertyName = "";
+
+            var options = rule.Configure(x => propertyName = x.GetDisplayName(null));
+
+            if (!string.IsNullOrEmpty(propertyName))
+                options.WithName(propertyName);
+
+            options.WithErrorCode($"{RuleTypeConst.Prefix}.{type}");
```

Comment:
thought: we're gonna need to use the same extension method `WithErrorCode` in #8, but I think we can use it in both scenarios. We can set some predefined words as ErrorCodes to change the response status code.

Reply:
Nice @luizhlelis, make sense.
We can use a new Enum.ErrorsStatusCode in this lib, creating a new method like "WithStatusCode(Enum.ErrorsStatusCode)".